### ISSUE TITLE
Valid random `RistrettoPoint` generation test

### ISSUE THREAD

**Issue by CPerezz** [dalek-cryptography/curve25519-dalek#288]
I was checking the tests for random `RistrettoPoint` generation, and I saw this:
```rust
#[cfg(feature = "rand")]
    #[test]
    fn random_is_valid() {
        let mut rng = OsRng::new().unwrap();
        for _ in 0..100 {
            let P = RistrettoPoint::random(&mut rng);
            // Check that P is on the curve
            assert!(P.0.is_valid());
            // Check that P is in the image of the ristretto map
            P.compress();
        }
    }
```

Since calling 
https://github.com/dalek-cryptography/curve25519-dalek/blob/cf03d39f0fc3e1c625b9f1e9be0473758b324526/src/ristretto.rs#L1319
will never cause the test to fail, why is it inside of the test?

I mean. Maybe it will make sense if we `compress` and then `decompress` `P` in order to check that it was a valid `RistrettoPoint` (letting `decompress` trigger a failure by returning `None` on an invalid `CompressedRistretto` decompression.

So is there any reason to call `compress` there?

**Comment by hdevalence** [comment]
I believe that at the time that the test was added, there was a `debug_assert!` on the return value of the invsqrt inside of the `compress` implementation, which was subsequently cleaned up.  So the whole test could be removed.

**Comment by CPerezz** [comment]
The whole test? Maybe it makes sense to leave it in order to demonstrate that whatever point obtained using `RistrettoPoint::random()` is a valid `EdwardsPoint` (satisfies the curve eq).

Also, to demonstrate that is suitable for being wrapped as a `RistrettoPoint`, we could call `decompress()` after the compression. This has to succeed always if the point is a valid representative. 

The only thing that I have not clear is if it has to be checked that the obtained points have order `L` or not (Cofactor 1) (Since there's nothing said about this on the docs if I'm not wrong).

**Comment by isislovecruft** [comment]
The whole test should probably be removed. It's not possible to obtain a `RistrettoPoint` that is not in the prime-order subgroup. Additionally, there's no point in demonstrating that "the [Ristretto] point is a valid `EdwardsPoint`," as it is well-defined irrespective of its underlying, internal, curve representation.

**Comment by isislovecruft** [comment]
Fixed in #302.
